### ISSUE TITLE
Feature request: Cross-entropy error function

### ISSUE THREAD

**Issue by tvedebrink** [tensorflow/playground#175]
In `nn.ts` it seems as only quadratic loss is supported in the playground? The definition of the `Error` class seems to indicate that one could implement the cross-entropy error function rather easy. With `target` being $t$ and $o$ for `output` The cross-entropy is given as
$$f(o,t) = t\log(o) + (1-t)\log(1-o),$$
where the derivative with respect to $o$ is 
$$f'(o,t) = \frac{\partial}{\partial o} t\log(o) + (1-t)\log(1-o) = \frac{t}{o} - \frac{1-t}{1-o} = \frac{t-o}{o(1-o)}$$
In this case, is my suggestion below correct?
```
public static CROSSENTROPY: ErrorFunction = {
    error: (output: number, target: number) =>
               target*log(output) + (1-target)*log(1-output),
    der: (output: number, target: number) => (target - output)/(output*(1-output))
  };
```
In the `playground.ts` script I would then change all references to `SQUARE` to `CROSSENTROPY` or make it possible to select the error function by drop-down menu?
